src/backend/routers/student.ts

```diff
@@ -35,6 +35,74 @@ export const student = router({
       return result;
     }),
 
+  getStudentDetailById: authenticatedProcedure
+    .input(z.object({ student_id: z.string().uuid() }))
+    .query(async (req) => {
+      const { student_id } = req.input;
+
+      const rows = await req.ctx.db
+        .selectFrom("student")
+        .innerJoin("iep", "iep.student_id", "student.student_id")
+        .innerJoin("goal", "goal.iep_id", "iep.iep_id")
+        .innerJoin("subgoal", "subgoal.goal_id", "goal.goal_id")
+        .where("student.student_id", "=", student_id)
+        .select([
+          "student.student_id",
+          "first_name",
+          "last_name",
+          "student.email",
+          "assigned_case_manager_id",
+          "grade",
+          "iep.iep_id",
+          "case_manager_id",
+          "start_date",
+          "end_date",
+          "iep.created_at",
+          "goal.goal_id",
+          "goal.description as goal_description",
+          "category",
+          "subgoal.subgoal_id",
+          "subgoal.description as subgoal_description",
+          "instructions",
+          "target_max_attempts",
+        ])
+        .execute();
+
+      // return rows as an array of objects which in turn contain objects representing tables
+      // [{{student, iep, goal, subgoal}}, {{student, iep, goal, subgoal}}, ...]
+      const tableRows = rows.map(function (row) {
```

Comment:
I was actually planning on doing just that, especially after seeing your implementation of that on the Iep procedure that you wrote. As of right now, this procedure isn't called on the front end, so it's something of a work in progress that may be abandoned all together. To that point, when we were discussing this yesterday, it opened up the larger question of how we should proceed with this kind of deeply nested document where everything is appearing on one page. Should we make this nested request, keep everything in the parent state, and pass everything down as props? Would this also mean needing to set up context or passing down additional procedures as props? Or should we expect to continue with the current approach of every rendered component calling its own procedure to get its own data? We were going to discuss this in the meeting as well but I think it's good to give you a head start if you'd like to think about it advance, plus writing this out helped me organize our thoughts about it.

Reply:
I'm not an expert, so this is just my opinion. I don't think every component should make an API call, because one student page could potentially be making tens or hundreds of calls (multiple subgoals for multiple goals for multiple* ieps). It may not be necessary to set up context if the props are simple, but it depends on implementation and can definitely be useful. We should definitely have this conversation with the team.

Reply:
As per our conversation in the meeting, especially with @amantri, this procedure has been removed, and I'm marking this as resolved.